src/sync.rs

```diff
@@ -62,7 +62,7 @@ mod inner {
     }
 }
 
-#[cfg(not(loom))]
+#[cfg(not(all(loom, test)))]
```

Comment:
and this needs to be
```suggestion
#[cfg(not(all(loom, any(feature = "loom", test))))]
```